### PR TITLE
Added OpenMP configuration on MacOS

### LINKED ISSUE #740: Add OMP to OSX CI tests

### PR DESCRIPTION
**Pull request by tonydp03**
This pull request enables the usage of OpenMP (and installation) for macOS in the CI, according to the issue [#740](https://github.com/alpaka-group/alpaka/issues/740). It has been tested locally on my Macbook Pro 2017.
The OpenMP5 backend still doesn't work on MacOS though (not fully supported).

### CODE REVIEW

File: script/before_install.sh [hunk 1]
```diff
@@ -165,4 +174,4 @@ then
             fi
         fi
     fi
-fi
\ No newline at end of file
+fi
```
Comment (bernhardmgruber) [inline script/before_install.sh]: Consider reverting this unnecessary whitespace change.

**Review by bernhardmgruber — DISMISSED** [review]

File: script/before_install.sh (hunk 1 above)
Comment (bernhardmgruber) [reply script/before_install.sh]: Oh wait, you added a new line in the end. Well, I guess this is less of an issue then. But still, try to avoid whitespace changes in areas of the code you did not touch.

**Review by bernhardmgruber — DISMISSED** [review]

**Review by bernhardmgruber — APPROVED** [review]

**Review by psychocoderHPC — APPROVED** [review]

**Comment by psychocoderHPC** [comment]
Decided in the developer VC to merge this before the CI passed.
